src/lib.rs

```diff
@@ -239,6 +239,64 @@ impl<'a, T> IntoIterator for &'a mut NonEmpty<T> {
     }
 }
 
+impl<T> NonEmpty<T> {
+    /// Removes the specified range from the vector in bulk, returning the removed items as an iterator.
+    /// # Panics
+    /// If the range specified would remove all elements from the vector. There must be at least 1 element left over.
```

Comment:
I think that if we want to adhere to rust's error handling philosophy, a panic is actually better here. If someone calls `v.drain(..)`, then they're asking "remove all the elements from this non-empty iterator". This is a nonsense question, so naturally we panic.  

We can see this same philosophy in the `Index` traits, which work the same way. They infallibly return their values when the programmer does the right thing, but they panic when the programmer asks a nonsense question, like for an out-of-bounds value.  

And in most cases, `drain` will get called with literal values, so it should be trivial for the programmer to avoid this panic. If we want to add safety for the few cases when people use runtime values, we could add another `checked_drain` method, akin to `slice::get` or `i32::checked_add`